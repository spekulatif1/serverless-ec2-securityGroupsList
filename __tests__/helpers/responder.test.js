'use strict'

const {
	responder
} = require('../../helpers');

describe('responder', () => {
	test('responder.generates.HTTP.200.response', (done) => {
		const succesPayload = [
			{
				"type": "articles",
				"id": "1",
				"attributes": {
					"title": "JSON:API paints my bikeshed!"
				}
			}
		];

		const successResponse = responder.buildSuccessResponse(succesPayload);
		expect(successResponse).toMatchSnapshot();

		done();
	});

	test('responder.generates.HTTP.500.response', (done) => {
		const errorMessage = 'testing.purposes';

		const errorResponse = responder.buildErrorResponse(errorMessage);
		expect(errorResponse).toMatchSnapshot();

		done();
	});

	test('responder.fallbacks.to.HTTP.500.with.not.defined.error.code', (done) => {
		const errorMessage = 'testing.with.not.defined.error.code';
		const notDefinedErrorCode = 'NotDefinedErrorCode';

		const errorResponse = responder.buildErrorResponse(
			errorMessage,
			notDefinedErrorCode
		);

		expect(errorResponse).toMatchSnapshot();

		done();
	});
});
