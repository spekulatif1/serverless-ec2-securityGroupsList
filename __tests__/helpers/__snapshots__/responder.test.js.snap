// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responder responder.fallbacks.to.HTTP.500.with.not.defined.error.code 1`] = `
Object {
  "body": "{\\"errors\\":[{\\"detail\\":\\"testing.with.not.defined.error.code\\",\\"title\\":\\"Internal Server Error\\"}]}",
  "statusCode": 500,
}
`;

exports[`responder responder.generates.HTTP.200.response 1`] = `
Object {
  "body": "{\\"data\\":[{\\"type\\":\\"articles\\",\\"id\\":\\"1\\",\\"attributes\\":{\\"title\\":\\"JSON:API paints my bikeshed!\\"}}]}",
  "statusCode": 200,
}
`;

exports[`responder responder.generates.HTTP.500.response 1`] = `
Object {
  "body": "{\\"errors\\":[{\\"detail\\":\\"testing.purposes\\",\\"title\\":\\"Internal Server Error\\"}]}",
  "statusCode": 500,
}
`;
